modules/packages/LinearAlgebra.chpl

```diff
@@ -433,6 +433,30 @@ proc eye(Dom: domain(2), type eltType=real) {
 // Matrix Operations
 //
 
+
+/* Sets the value of a diagonal in a dense matrix.
+
+    ``p > 0``, represents an upper diagonal starting
+    from the ``p``th column, ``p == 0`` represents the main 
+    diagonal, ``p < 0`` represents a lower diagonal starting
+    from the ``-p``th row. ``p`` is 0-indexed.
+*/
+proc setDiag (ref X: [?D] ?eltType, in p: int = 0, val: eltType = 0) 
```

Comment:
nit: We use `k` as the diagonal-offset argument in other diagonal-related functions like `diag()` and `tril()`. Would be more consistent to stick with `k`.

Reply:
Yes the performance results are up to date